src/full/Agda/Utils/Parser/ReadP.hs

```diff
@@ -102,8 +102,8 @@ instance Applicative (P t) where
 instance Monad (P t) where
   return = pure
 
-  (Get f)      >>= k = Get (\c -> f c >>= k)
-  (Look f)     >>= k = Look (\s -> f s >>= k)
+  (Get f)      >>= k = Get (f >=> k)
```

Comment:
This module hasn't seen any significant changes for a decade.  You are treading the ancient code halls of Agda.  Draw your hat and show respect! :D  
But me too, I sometimes feel the lust of vandalism and do little modernizations of the code here and there.

Reply:
It is also a good way to learn how Agda works internally. :-)